hare/eligibility/oracle.go

```diff
@@ -30,33 +39,66 @@ type VerifierFunc = func(msg, sig, pub []byte) (bool, error)
 
 // Oracle is the hare eligibility oracle
 type Oracle struct {
-	beacon            valueProvider
-	activeSetProvider activeSetProvider
-	vrfSigner         Signer
-	vrfVerifier       VerifierFunc
-	layersPerEpoch    uint16
+	beacon         valueProvider
+	getActiveSet   activeSetFunc
+	vrfSigner      Signer
+	vrfVerifier    VerifierFunc
+	layersPerEpoch uint16
+	cache          casher
+	cfg            eCfg.Config
 	log.Log
 }
 
-// Returns the relative Layer that w.h.p. we have agreement on its view
-// safe is defined to be k layers prior to the provided Layer
-func safeLayer(layer types.LayerID) types.LayerID {
-	if layer > k { // assuming genesis is zero
-		return layer - k
+// Returns the relative layer id that w.h.p. we have agreement on its contextually valid blocks
+// safe layer is defined to be the confidence interval layers prior to the provided Layer
```

Comment:
"We define safe layer to be k layers ago, where k is our `confidence parameter' "